### ISSUE TITLE
Document the various code generation tools used by CPython

### ISSUE THREAD

**Issue by pxeger** [python/devguide#805]
**Describe the enhancement or feature you'd like**

There should be some documentation about the various code generation tools used in CPython, for example:

- freeze_modules.py
- deepfreeze.py
- generate_global_objects.py

I don't think there needs to be a massive amount of documentation about these. Just a brief mention of their purposes, how to invoke them, and notes about their related `make` targets.

Related are the Parser generator etc., which are already documented under [Guide to CPython's Parser](https://devguide.python.org/parser/#how-to-regenerate-the-parser), but could also be mentioned. I think their existing docs could also be used as something of a template.

**Describe alternatives you've considered**

Argument clinic is already documented at https://docs.python.org/3/howto/clinic.html so maybe this documentation would be more appropriate in the main Python repo's docs? I'll defer to the devguide maintainers on this.

**Additional context**

Started from a conversation at https://github.com/python/cpython/pull/30941#issuecomment-1023501086

CC @ericisnowcurrently

**Comment by erlend-aasland** [comment]
> I don't think there needs to be a massive amount of documentation about these. Just a brief mention of their purposes, how to invoke them, and notes about their related `make` targets.

I agree. A short how-to-use-the-various-generation-tools section would be helpful to new (and existing) contributors/devs. That includes Argument Clinic; the current how-to is huge and frightening; we should definitely add an introduction to the clinic CLI to the devguide.

**Comment by erlend-aasland** [comment]
See also https://discuss.python.org/t/where-should-the-argument-clinic-docs-live/30160/

**Comment by AA-Turner** [comment]
The Argument Clinic documentation now lives here!

A
